Library/ViewModels/PledgeCTAContainerViewViewModel.swift

```diff
@@ -0,0 +1,92 @@
+import KsApi
+import Prelude
+import ReactiveExtensions
+import ReactiveSwift
+
+public protocol PledgeCTAContainerViewViewModelInputs {
+  func configureWith(project: Project, user: User)
+}
+
+public protocol PledgeCTAContainerViewViewModelOutputs {
+  var buttonBackgroundColor: Signal<UIColor, Never> { get }
+  var buttonTitleText: Signal<String, Never> { get }
+  var rewardTitle: Signal<String, Never> { get }
+  var stackViewIsHidden: Signal<Bool, Never> { get }
+}
+
+public protocol PledgeCTAContainerViewViewModelType {
+  var inputs: PledgeCTAContainerViewViewModelInputs { get }
+  var outputs: PledgeCTAContainerViewViewModelOutputs { get }
+}
+
+public final class PledgeCTAContainerViewViewModel: PledgeCTAContainerViewViewModelType,
+  PledgeCTAContainerViewViewModelInputs, PledgeCTAContainerViewViewModelOutputs {
+  public init() {
+    let projectAndUser = self.projectAndUserProperty.signal.skipNil()
+    let project = projectAndUser.map { $0.0 }
+
+    let pledgeState = project
+      .map { pledgeStateButton(project: $0) }
+
+    let backingEvent = projectAndUser
+      .switchMap { project, user in
+        AppEnvironment.current.apiService.fetchBacking(forProject: project, forUser: user)
+          .materialize()
+    }
+    let backing = backingEvent.values()
+    let projectAndBacking = Signal.combineLatest(project, backing)
+
+    self.buttonTitleText = pledgeState.map { $0.buttonTitle }
+    self.buttonBackgroundColor = pledgeState.map { $0.buttonBackgroundColor }
+    self.stackViewIsHidden = pledgeState.map { $0.stackViewIsHidden }
+
+    self.rewardTitle = projectAndBacking
+      .map { (arg) -> String in
```

Comment:
Do we need the ambiguous `arg` here? Looks like the following works on my machine..

```swift
  .map { (project, backing) -> String in
```